### DISCUSSION TITLE
SDK requirements on a project using these packages?

### DISCUSSION THREAD

**Discussion by KalleOlaviNiemitalo** [dotnet/extensions#4736] [Q&A] [answered]
What are the lowest supported versions of MSBuild and Roslyn, for building a C# project that targets netstandard2.0 and uses 8.0.0 versions of Microsoft.Extensions.\* packages from this repository?

Is the compatibility covered by automated tests?

**Comment by joperezr** [comment, marked as answer]
Some of our packages contain generators, and those build against Roslyn 4.4.0, which means that if you want to build in visual studio, you'll want to be using Visual Studio 2022 17.4 or later (see [Roslyn and VS version mappings ](https://learn.microsoft.com/en-us/visualstudio/extensibility/roslyn-version-support?view=vs-2022)for more info).

For building from the command line, you will also need to have at least 17.4 MSBuild, which means that either a 7.0 or an 8.0 SDK should be sufficient.

> Is the compatibility covered by automated tests?

All of our tests and builds are against 8.0 SDKs, we don't currently have anything ensuring that 7.0 SDKs also work, but there shouldn't be a reason why this wouldn't work. If you have a 7.0 SDK and trying to consume a package and having issues, please let us know so that we can investigate.

**Reply by KalleOlaviNiemitalo** [reply]
What here requires MSBuild 17.4?  I didn't find any MSBuild tasks in this repository.

**Reply by RussKie** [reply]
https://github.com/dotnet/extensions/blob/f34d120d2654057a31dc96d7f86dc42629044472/eng/packages/General.props#L8-L11

**Reply by KalleOlaviNiemitalo** [reply]
Those packages do not contain any MSBuild tasks, either; they do not contain any `build` or `buildTransitive` directories. Even if they did, that version of Roslyn appears to reference MSBuild 16.5 rather than 17.4: <https://github.com/dotnet/roslyn/blob/1ce8866c9de4c2d67351ef6863699dee03b0804a/eng/Versions.props#L37>

**Reply by joperezr** [reply]
It's not just MSBuild tasks (or things in /build or /buildtransitive folders) what can depend on specific Roslyn versions. In our case, it's generators which are included in our packages. If you take a look at some of the packages like:

https://github.com/dotnet/extensions/blob/c8383902da237c8efd61a37808e444ac9b838f59/src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Microsoft.Extensions.Telemetry.Abstractions.csproj#L44-L45

You'll find that some do ship analyzers in them, and these analyzers depend on a specific version of Roslyn, which in turn requires a specific version of VS. I know above I said that there are requirements for MSBuild 17.4, but I mostly said that because this is the version of MSBuild which comes with the minium Roslyn version required by our generators.
